packages/arb-bridge-eth/contracts/validator/GasRefunder.sol

```diff
@@ -0,0 +1,118 @@
+        address indexed contractAddress,
+        uint256 gas,
+        uint256 reason
+    );
+    event Deposited(address sender, uint256 amount);
+    event Withdrawn(address initiator, address destination, uint256 amount);
+    event ContractAllowedSet(address indexed contractAddress, bool allowed);
+    event MaxRefundeeBalanceSet(uint256 balance);
+    event ExtraGasMarginSet(uint256 balance);
+
+    function setContractAllowed(address contractAddress, bool allowed) external onlyOwner {
+        allowedContracts[contractAddress] = allowed;
+        emit ContractAllowedSet(contractAddress, allowed);
+    }
+
+    function setMaxRefundeeBalance(uint256 newMax) external onlyOwner {
+        maxRefundeeBalance = newMax;
+        emit MaxRefundeeBalanceSet(newMax);
+    }
+
+    function extraGasMarginSet(uint256 newMargin) external onlyOwner {
+        extraGasMargin = newMargin;
+        emit ExtraGasMarginSet(newMargin);
+    }
+
+    receive() external payable {
+        emit Deposited(msg.sender, msg.value);
+    }
+
+    function withdraw(address payable destination, uint256 amount) external onlyOwner {
+        destination.transfer(amount);
+        emit Withdrawn(msg.sender, destination, amount);
+    }
+
+    function onGasSpent(address payable refundee, uint256 gasUsed) external override {
+        uint256 startGasLeft = gasleft();
+        require(allowedContracts[msg.sender], "NOT_ALLOWED_CONTRACT");
+
+        if (lastContractRefund[msg.sender] == block.number) {
+            // There was already a refund this block, don't refund further
+            emit RefundGasCostsDenied(refundee, msg.sender, gasUsed, 0);
+            return;
+        }
+        lastContractRefund[msg.sender] = block.number;
+
+        // Allow for up to a 2 gwei tip
+        uint256 estGasPrice = block.basefee + 2e9;
+        if (tx.gasprice < estGasPrice) {
+            estGasPrice = tx.gasprice;
+        }
+
+        // Cache these variables and retrieve them before measuring gasleft()
+        uint256 refundeeBalance = refundee.balance;
+        uint256 maxRefundeeBalanceCache = maxRefundeeBalance;
+
+        // Add in a bit of a buffer for the tx costs not measured with gasleft
+        gasUsed += startGasLeft + extraGasMargin;
+        // Split this up into two statements so that gasleft() comes after the storage load of extraGasMargin
+        gasUsed -= gasleft();
+
+        uint256 refundAmount = estGasPrice * gasUsed;
+        if (
+            maxRefundeeBalanceCache != 0 && refundeeBalance + refundAmount > maxRefundeeBalanceCache
+        ) {
+            if (refundeeBalance > maxRefundeeBalanceCache) {
+                // The refundee is already above their max balance
+                emit RefundGasCostsDenied(refundee, msg.sender, gasUsed, 1);
+                return;
+            } else {
+                refundAmount = maxRefundeeBalanceCache - refundeeBalance;
+            }
+        }
+
+        bool success = refundee.send(refundAmount);
```

Comment:
```suggestion
        bool success = refundee.call{ value: refundAmount }("");
```
https://consensys.github.io/smart-contract-best-practices/recommendations/#dont-use-transfer-or-send

Reply:
We could also potentially limit the gas sent in this call instead of just the evm `63/64`